### PR TITLE
DAOS-15638 tests: move cat_recovery test from pr to daily test

### PR DESCRIPTION
**Pull request by Nasf-Fan**
DAOS catastrophic recovery core tests may take more than 1 hour. Keep it in PR test set is too long for most of non-cat_recov PR. Let's move it to daily test set.

Add cat_recov tag for catastrophic recovery test cases.

Skip-nlt: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-small: true
Test-tag: cat_recov

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/recovery/cat_recov_core.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        Test Description:
+            Run daos_test -F
+
+        Use cases:
+            Core tests for daos catastrophic recovery
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium,provider
+        :avocado: tags=recovery,test_cat_recovery,cat_recov_core
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/cat_recov_core.py:27]: We shouldn't use tags that start with `test_` unless it's the name of the test method. Can we just call it `cat_recovery`? 

To be consistent, I think `cat_recov_core` should be `cat_recovery_core`

Also, do we really need three separate tags? `recovery`, `cat_recovery`, `cat_recovery_core`?

The test method name tag is already unique amongst all tests: `test_daos_cat_recov_core`


And all the tests being tagged with `test_cat_recovery` now already have `recovery`

File: src/tests/ftest/recovery/cat_recov_core.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# change host names to your reserved nodes, the
+# required quantity is indicated by the placeholders
+hosts:
+  test_servers: 4
+# Note that subtests below can set their own timeout so this
+# should be a general average of all tests not including outliers
+# (I'm looking at you "rebuild tests")
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/cat_recov_core.yaml:7]: We shouldn't copy these comments
```suggestion
hosts:
  test_servers: 4
```

File: src/tests/ftest/recovery/container_cleanup.py [hunk 3]
```diff
@@ -34,9 +34,9 @@ def test_container_label_inconsistency(self):
 
         Jira ID: DAOS-12289
 
-        :avocado: tags=all,pr
-        :avocado: tags=vm
-        :avocado: tags=recovery,container_cleanup
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=recovery,test_cat_recovery,container_cleanup
         :avocado: tags=ContainerCleanupTest,test_container_label_inconsistency
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/container_cleanup.py:40]: Intentionally moved from VM to HW?

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: src/tests/ftest/recovery/cat_recov_core.py (hunk 1 above)
Comment (Nasf-Fan) [reply src/tests/ftest/recovery/cat_recov_core.py:27]: I will unify the tag as "cat_recov", remove "test_cat_recovery" and "cat_recovery_core".

File: src/tests/ftest/recovery/container_cleanup.py (hunk 3 above)
Comment (Nasf-Fan) [reply src/tests/ftest/recovery/container_cleanup.py:40]: Yes, CR tests are relative slow, run in VM will be more slow, let's test in HW.

File: src/tests/ftest/recovery/container_cleanup.py [hunk 4]
```diff
@@ -34,9 +34,9 @@ def test_container_label_inconsistency(self):
 
         Jira ID: DAOS-12289
 
-        :avocado: tags=all,pr
-        :avocado: tags=vm
-        :avocado: tags=recovery,container_cleanup
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/container_cleanup.py:38]: If moving to HW, we should change the config to `storage: auto`
https://github.com/daos-stack/daos/blob/1d3929216fe5d1cbae325887c0e2487fec5ed514/src/tests/ftest/recovery/container_cleanup.yaml#L13-L16

File: src/tests/ftest/recovery/pool_cleanup.py [hunk 5]
```diff
@@ -34,9 +34,9 @@ def test_corrupt_label_ms(self):
 
         Jira ID: DAOS-11741
 
-        :avocado: tags=all,pr
-        :avocado: tags=vm
-        :avocado: tags=recovery,pool_cleanup
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/pool_cleanup.py:38]: And here
https://github.com/daos-stack/daos/blob/1d3929216fe5d1cbae325887c0e2487fec5ed514/src/tests/ftest/recovery/pool_cleanup.yaml#L13-L16

**Review by daltonbohning — COMMENTED** [review]
Sorry for being picky, but we have _**A LOT**_ of arbitrary tags in ftest so I'm trying to make sure we don't make this more complicated if there isn't a real benefit. Here are some guidelines, and please let me know what makes sense from your perspective as a developer in this area.


It looks like all catastrophic recovery tests are tagged with `recovery` and `cat_recov`. Do we need both or could we just use `recovery`? That way it also corresponds to the `ftest/recovery` directory. But if you do want both to split tests in sub groups, can we make it `cat_recov->cat_recovery` so it follows similar naming convention as `recovery`?

Another idea is that if `cat_recovery` makes more sense than `recovery`, we could rename `ftest/recovery` to `ftest/cat_recovery` and keep ONLY the `cat_recovery` tag. If we go this route, I can do this myself on a separate PR.

File: src/tests/ftest/recovery/container_cleanup.py (hunk 3 above)
Comment (daltonbohning) [reply src/tests/ftest/recovery/container_cleanup.py]: This is a tricky decision. HW is faster than VMs, but VMs are cheaper and we don't have as much contention for VM resources. HW Medium has the most contention and takes the most time overall. So unless the tests are VMs are significantly slower than on HW, we should keep them on VMs

Comment (Nasf-Fan) [reply src/tests/ftest/recovery/container_cleanup.py]: We have many cat_recovery test cases under the recovery directory, most of with tag "hw". It is unnecessary to split them into "VM" and "HW". I prefer to unify them.

**Comment by Nasf-Fan** [comment]
> Sorry for being picky, but we have _**A LOT**_ of arbitrary tags in ftest so I'm trying to make sure we don't make this more complicated if there isn't a real benefit. Here are some guidelines, and please let me know what makes sense from your perspective as a developer in this area.
> 
> It looks like all catastrophic recovery tests are tagged with `recovery` and `cat_recov`. Do we need both or could we just use `recovery`? That way it also corresponds to the `ftest/recovery` directory. But if you do want both to split tests in sub groups, can we make it `cat_recov->cat_recovery` so it follows similar naming convention as `recovery`?
> 
> Another idea is that if `cat_recovery` makes more sense than `recovery`, we could rename `ftest/recovery` to `ftest/cat_recovery` and keep ONLY the `cat_recovery` tag. If we go this route, I can do this myself on a separate PR.

There are two kinds of test cases under "recovery" directory, one part is for "ddb", the others are for kinds of distributed consistency functionalities. All of them have tag "recovery". So we need another tag to distinguish them. That is why I keep both "recovery" and "cat_recov".

File: src/tests/ftest/recovery/container_cleanup.py (hunk 4 above)
Comment (Nasf-Fan) [reply src/tests/ftest/recovery/container_cleanup.py:38]: I will fix that.

File: src/tests/ftest/recovery/container_cleanup.py (hunk 3 above)
Comment (daltonbohning) [reply src/tests/ftest/recovery/container_cleanup.py]: > It is unnecessary to split them into "VM" and "HW"

I disagree. If a test can run on VMs, we should run it on VMs to save HW resources. Moving tests to HW contributes to CI slowness

**Comment by daltonbohning** [comment]
> There are two kinds of test cases under "recovery" directory, one part is for "ddb", the others are for kinds of distributed consistency functionalities. All of them have tag "recovery". So we need another tag to distinguish them. That is why I keep both "recovery" and "cat_recov".

I updated this page with those two tags:
https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags#[inlineExtension]List-of-Feature-Tags
And please feel free to update in the future as needed.

**Comment by Nasf-Fan** [comment]
> > There are two kinds of test cases under "recovery" directory, one part is for "ddb", the others are for kinds of distributed consistency functionalities. All of them have tag "recovery". So we need another tag to distinguish them. That is why I keep both "recovery" and "cat_recov".
> 
> I updated this page with those two tags: [https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags#[inlineExtension]List-of-Feature-Tags](https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags#%5BinlineExtension%5DList-of-Feature-Tags) And please feel free to update in the future as needed.

Thanks for that.

File: src/tests/ftest/recovery/container_cleanup.py (hunk 3 above)
Comment (daltonbohning) [reply src/tests/ftest/recovery/container_cleanup.py]: > some time-consuming test cases run on "VM" may need 2 hours, but run on "HW" may just take 30 minutes

Yes, I agree that could be a reason to move to HW. And we have done similar in the past.
Looking at a master run, where these tests are on VM:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/9652/testReport/FTEST_recovery/

And yours, where they are on HW:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14161/3/testReport/FTEST_recovery/

They are indeed faster on HW.

Not a blocker, but ideally we should reduce the test timeouts if they are running faster now.

**Review by daltonbohning — COMMENTED** [review]
Changes LGTM, but now that these aren't `pr` tests, we need to run with
```
Test-tag: cat_recov
```
FWIW, I'm very close to automating that

**Comment by Nasf-Fan** [comment]
> Changes LGTM, but now that these aren't `pr` tests, we need to run with
> 
> ```
> Test-tag: cat_recov
> ```
> 
> FWIW, I'm very close to automating that

Pushed another commit with more fixes and the "Test-tag: cat_recov".

**Review by daltonbohning — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Pull request merged**